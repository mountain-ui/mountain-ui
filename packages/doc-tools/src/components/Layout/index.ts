export { default as Box } from './Box';
export * from './Box';
export * from './Container';
