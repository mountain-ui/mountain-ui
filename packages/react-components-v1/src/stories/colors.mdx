import { Meta, ColorPalette, ColorItem } from '@storybook/blocks';

<Meta title='Design System/Colors' />

# Colors

For the color palette, we inherit the [Tailwind color palette](https://tailwindcss.com/docs/customizing-colors), adding some custom colors based on out needs.

<ColorPalette>
  <ColorItem title='Light' colors={['hsl(0, 0%, 98%)']} />
  <ColorItem title='Dark' colors={['hsl(210, 14%, 9%)']} />
</ColorPalette>
