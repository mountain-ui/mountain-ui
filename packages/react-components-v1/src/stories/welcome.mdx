import { Meta } from '@storybook/blocks';

<Meta title='Design System/Welcome' />

<h1 className='mb-2 text-h1 text-green-400'>Mountain UI 🏔</h1>
<p mb={2}>
  This tool helps us explore, text and document all the components from the Mountain UI design
  system.
</p>
<p mb={2}>
  It is divided into more stories, you can find them on the left navigation sidebar. Start clicking
  a story to play around with the library components.
</p>
