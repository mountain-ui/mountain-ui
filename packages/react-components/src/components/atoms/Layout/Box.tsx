import mnt from 'react-mnt';

/**
 * Box used to define Layout components
 */
export const MntBox = mnt('div')``;

export type MntBoxProps = React.ComponentPropsWithRef<typeof MntBox>;
