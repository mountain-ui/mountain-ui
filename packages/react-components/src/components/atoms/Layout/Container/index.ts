export { MntContainer } from './Container';
export type { MntContainerProps, MntContainerSize } from './Container';
