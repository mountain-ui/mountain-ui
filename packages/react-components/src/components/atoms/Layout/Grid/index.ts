export { MntGrid } from './Grid';
export type { MntGridProps, MntGridColumns } from './Grid';
