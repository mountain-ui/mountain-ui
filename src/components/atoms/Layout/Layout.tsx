import styled from 'styled-components';
import { grid } from 'styled-system';

import { BaseElement, BaseElementProps } from '../../../BaseElement';

export type LayoutProps = BaseElementProps

/**
 * Container component
 */
export const Container = styled(BaseElement) <LayoutProps>``
Container.defaultProps = {
  maxWidth: ['mobile', 'tablet', 'desktop'],
  px: 3,
  mx: 'auto',
}
Container.displayName = 'Container'

/**
 * Box component
 */
export const Box = BaseElement
Box.displayName = 'Box'

/**
 * Grid component
 */
export const Grid = styled(BaseElement)`
  ${grid}
  display: grid;
`;
Grid.displayName = 'Grid'

export default BaseElement