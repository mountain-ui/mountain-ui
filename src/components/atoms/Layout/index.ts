export { default as Box } from './Box';
export * from './Box';
export * from './Container';
export * from './Grid';
