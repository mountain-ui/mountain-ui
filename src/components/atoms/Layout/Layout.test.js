import React from 'react';
import initStoryshots from '@storybook/addon-storyshots';
import { render } from '@testing-library/react';

import { ContainerStory, GridStory } from './Layout.stories';

initStoryshots();

describe('Layout', () => {
  describe('Container component', () => {
    it('should render correctly on mount', () => {
      const { container } = render(<ContainerStory {...ContainerStory.args} />);
      expect(container).toBeInTheDocument();
    });
  });

  describe('Grid component', () => {
    it('should render correctly on mount', () => {
      const { container } = render(<GridStory {...GridStory.args} />);
      expect(container).toBeInTheDocument();
    });
  });
});
