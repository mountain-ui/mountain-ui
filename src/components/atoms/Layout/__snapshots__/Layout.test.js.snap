// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Button Outline 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <button
    className="sc-bdfBwQ sc-hKgILt  iGvDCY"
  >
    Click
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Primary 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <button
    className="sc-bdfBwQ sc-hKgILt  hxYeM"
  >
    Click
  </button>
</div>
`;

exports[`Storyshots Atoms/Layout Container 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <div
    className="sc-bdfBwQ sc-gsTCUz bFhuNW kNNlrj"
    height="100vh"
  >
    This is a responsive container
  </div>
</div>
`;

exports[`Storyshots Atoms/Layout Grid 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <div
    className="sc-bdfBwQ sc-dlfnbm bFhuNW dYmuJF"
    height="100vh"
  >
    This is a responsive Grid
  </div>
</div>
`;

exports[`Storyshots Atoms/Typography/Heading Heading 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <h1
    className="sc-bdfBwQ sc-eCssSg sc-jSgupP gYDVUp kVzVMq heeOhy"
    color="text.primary"
  >
    Mountain UI library
  </h1>
</div>
`;

exports[`Storyshots Atoms/Typography/Paragraph Paragraph 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <p
    className="sc-bdfBwQ sc-eCssSg sc-gKsewC sc-iBPRYJ  kVzVMq"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
  </p>
</div>
`;

exports[`Storyshots Atoms/Typography/Text Text 1`] = `
<div
  className="sc-bdfBwQ sc-gsTCUz  kNNlrj"
>
  <span
    className="sc-bdfBwQ sc-eCssSg sc-gKsewC  kVzVMq"
  >
    Mountain UI library
  </span>
</div>
`;
